src/main/java/uk/gov/justice/probation/courtcaseservice/controller/CaseSearchController.java

```diff
@@ -31,9 +31,10 @@ public class CaseSearchController {
     public CaseSearchController(final CaseSearchService caseSearchService) {
         this.caseSearchService = caseSearchService;
     }
-    @Operation(description = "Search cases by CRN")
-    @GetMapping(value = "/search", produces = APPLICATION_JSON_VALUE)
-    public @ResponseBody CaseSearchResult searchByCrn(@RequestParam("crn") String crn) {
-        return caseSearchService.searchByCrn(crn);
+
+    @Operation(description = "Search cases by CRN or name")
+    @GetMapping(value = "/search", params = {"term"}, produces = APPLICATION_JSON_VALUE)
+    public @ResponseBody CaseSearchResult searchCases(@RequestParam("term") String term) {
```

Comment:
Just aligning with the end to end journey. On the front end it is the same text field and the user can enter either CRN or name in it. So this prediction of whether it is CRN or name has to exist either in prepare-a-case or court-case-service and it is better the backend does the heavy lifting.

Reply:
Ideally, we should have separate endpoints, if that's not possible then pass the 'type' as well.

Reply:
I don't think it is ideal to have search endpoints for each type. search endpoints have always been solo and general ones with filters/types. I don't think it would be ideal to to create a separate endpoint for each search filter/type, say we have 10 filters/types?. I followed the same principle, instead the endpoint is clever enough to derive the filter itself. nDelius works this way

Reply:
Agree with the filter search on this. just pass the type will do it.

Reply:
Updated, thanks